File: python/signup.py
#! /usr/bin/python3

from urllib import response
import requests
import getpass

response = ""


def login(username, email, password):
    payload = {
        'username': username,
        'pass': password,
        'email': email
    }
    response = requests.post("http://api.com/auth/signup", data=payload)
    print(response.text)
    if (response.status_code == 200):
        return 200
    elif (response.status_code == 400):
        return 400
    elif (response.status_code == 406):
        return 406
    else:
        pass


while True:

    u = input("Enter Username :")

    if u in ['']:
        print("Type Username Properly")
    else:
        break
while True:
    e = input("Enter Email : ")

    if e in ['']:
        print("Type Username Properly")
    else:
        break

while True:
    p = getpass.getpass("Enter Password :")

    if p in ['']:
        print("Type Username Properly")
    else:
        break

if (login(u, e, p) == 200):
    print("Account Created Successfully")
elif (login(u, e, p) == 406):
    print("Not Acceptable")
else:
    print("Bad Request")

File: apis/auth/signup.php
<?php
require_once($_SERVER['DOCUMENT_ROOT'] . "/api/Rest.api.php");
require_once($_SERVER['DOCUMENT_ROOT'] . "/api/lib/Database.class.php");
require_once($_SERVER['DOCUMENT_ROOT'] . "/api/lib/Signup.class.php");
${basename(__FILE__, '.php')} = function () {
    try {
        if ($this->get_request_method() == "POST") {
            if (isset($this->_request['pass']) and isset($this->_request['username']) and isset($this->_request['email'])) {
                $email = $this->_request['email'];
                $pass = $this->_request['pass'];
                $username = $this->_request['username'];
                $s = new Signup($username, $pass, $email);
                $data = [
                    "Message" => "User Created Successfully",
                    "Insert ID" => $s->getInsertID(),
                ];
                $data = $this->json($data);
                $this->response($data, 200);
            } else {
                $data = [
                    "Error" => "Not Acceptable",
                ];
                $data = $this->json($data);
                $this->response($data, 406);
            }
        } else {
            $data = [
                "Error" => "Bad Request",
            ];
            $data = $this->json($data);
            $this->response($data, 400);
        }
    } catch (\Exception $e) {
        $data = [
            "msg" => $e->getMessage(),
        ];
        $data = $this->json($data);
        $this->response($data, 400);
    }
};
